test/inductor/test_fp8.py

```diff
@@ -346,6 +368,67 @@ def ln_fp8(x: Tensor, scale: Tensor, amax_buffer: Tensor):
         )
 
 
+@instantiate_parametrized_tests
+class TestFP8Lowering(TestCase):
+    @unittest.skipIf(TEST_WITH_ROCM, "FP8 is not supported on ROCM")
+    @unittest.skipIf(not SM90OrLater, "FP8 is only supported on H100+")
+    @parametrize("dtype", (torch.bfloat16,))
+    @parametrize("shape", ("16,16,32", "1024,1024,512"))
+    @parametrize("has_bias", (False, True))
+    def test_tensorwise_scaling(self, dtype: torch.dtype, shape: str, has_bias: bool):
+        device = "cuda"
+        dtype_float8 = torch.float8_e4m3fn
+        use_fast_accum = True
+
+        shape = [int(dim) for dim in shape.split(",")]
+        M, K, N = shape  # Matmul Y = X [M, K] x W [N, K]
+        x = torch.rand(M, K, dtype=dtype, device=device)
+        w = torch.rand(N, K, dtype=dtype, device=device)
+
+        bias = None
+        if has_bias:
+            bias = torch.rand(N, device=device, dtype=torch.bfloat16)
+
+        # quantize weight (prior to inference)
+        weight_amax = torch.max(torch.abs(w))
+        weight_scale = _amax_to_scale(weight_amax, dtype_float8, w.dtype)
+        w_t_fp8 = _to_fp8_saturated(w * weight_scale, dtype_float8).t()
+        w_inverse_scale = weight_scale.reciprocal()
+
+        def linear(x, w_t_fp8, w_inverse_scale, bias):
+            # quantize input x
+            amax = torch.max(torch.abs(x))  # tensor-wise
+            scale = _amax_to_scale(amax, dtype_float8, x.dtype)
+            x_fp8 = _to_fp8_saturated(x * scale, dtype_float8)
+            x_inverse_scale = scale.reciprocal()
+
+            y = torch._scaled_mm(
+                x_fp8,
+                w_t_fp8,
+                x_inverse_scale,
+                w_inverse_scale,
+                bias,
+                out_dtype=dtype,
+                use_fast_accum=use_fast_accum,
+            )
+            return y
+
+        y_eager = linear(
+            x,
+            w_t_fp8,
+            w_inverse_scale,
+            bias,
+        )
+        linear_compiled = torch.compile(linear, backend="inductor", mode="max-autotune")
+        y_compiled = linear_compiled(
+            x,
+            w_t_fp8,
+            w_inverse_scale,
+            bias,
+        )
+        torch.testing.assert_close(y_eager, y_compiled, rtol=5e-1, atol=5e-1)
```

Comment:
does this match how other matmul codegen cases in inductor are tested?

Reply:
`rtol=5e-1, atol=5e-1` is the most strict used elsewhere in this file e.g. `test_valid_cast()`. In the Triton matmul [tutorial](https://triton-lang.org/main/getting-started/tutorials/03-matrix-multiplication.html#sphx-glr-getting-started-tutorials-03-matrix-multiplication-py), for fp8 it tested `atol=0.125, rtol=0` for M, N, K = 512 as a reference. Let me know if you have concerns!

Reply:
the float8 test in the triton tutorial is comparing PyTorch bf16 matmul to triton float8 matmul.  The comparison here is between eager mode float8 matmul and compiled float8 matmul - should that be tested with a higher tolerance?

Reply:
You're right, let me check the actual eager vs compiled difference for some shapes to see what's a good strict tolerance.

Reply:
@vkuzo updated the tolerance for each after checking the actual differences for the shapes tested. I initially read `5e-1` as `1e-5` in my head which was not helpful...